lib/managers/files.js

```diff
@@ -1344,6 +1375,53 @@ Files.prototype.getRepresentationInfo = function(fileID, representationType, cal
 		})
 		.asCallback(callback);
 };
+
+/**
+ * Get the contents of a representation of a file, e.g, the binary content of an image or pdf.
+ *
+ * API Endpoint: '/files/:fileID?fields=representations'
+ * Method : GET
+ *
+ * @param {string} fileID The file ID to get the representation of
+ * @param {string} representationType The X-Rep-Hints type to request
+ * @param {Object} [options] Optional parameters
+ * @param {string} [options.assetPath] Asset path for representations with multiple files
+ * @param {Function} [callback] Passed a stream over the representation contents if successful
+ * @returns {Promise<Readable>} A promise resolving to a stream over the representation contents
+ */
+Files.prototype.getRepresentationContent = function(fileID, representationType, options, callback) {
+
+	options = Object.assign({ assetPath: '' }, options);
+
+	return this.getRepresentationInfo(fileID, representationType)
+		.then(reps => {
+
+			var repInfo = reps.entries.pop();
+			if (!repInfo) {
+				throw new Error('Could not get information for requested representation');
+			}
+
+			switch (repInfo.status.state) {
+
+			case 'success':
+			case 'viewable':
+				return repInfo.content.url_template;
+			case 'error':
+				throw new Error('Representation had error status');
+			case 'none':
+			case 'pending':
+				return pollRepresentationInfo(this.client, repInfo.info.url);
```

Comment:
This should have a delay as well, not the other comment about extracting into a common function.

Reply:
I don't think we want to inject unnecessary latency here — if the status was `"none"`, we need to make the request to the info URL immediately to kick off generation of the representation.

Reply:
Ah, yes. The delay is only beneficial if the status is pending, but it's not required. Thanks for pointing that out.